src/MassTransit/Context/SendContextProxy.cs

```diff
@@ -100,6 +100,12 @@ public Uri FaultAddress
             set => _context.TimeToLive = value;
         }
 
+        public DateTime? SentTime
+        {
+            get => _context.SentTime;
+            set => _context.SentTime = value;

```

Comment:
Set should not be allowed through a proxy.

Reply:
What about the other sets? The proxy implements the SendContext interface indirectly (through PublishContext interface). Removing set would break the implementation. I believe the ConsumeContext is different, given the set actually happens when deserializing the envelope/message. And setting SentTime at instantiation seems too early, wouldn't you agree? Suggestions?